### PR TITLE
add a `format` task to tool/grind.dart

### PR DESCRIPTION
**Pull request by zoechi**
With the now formatted Dart code it would make sense to add a `format` task to keep the code formatted.
I don't know if you want it added to the default task.

I prefer passing a concrete list of directories to `dartfmt` because for example with `.` `dartfmt` also formats the `build` directory, if there is any, which then also contains a copy of all dependencies which can make it quite a lengthy but pointless task.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
   use `RunOptions.workingDirectory` instead.
 - Added `PubApp.runAsync`.
 - In `Dart.run`, deprecate the vmNewGenHeapMB and vmOldGenHeapMB options.
+- Support to pass lists of files or directories to `DartFmt.format()`
```
Comment (devoncarew) [inline changelog.md]: ==> `Support for passing`

File: lib/grinder_sdk.dart [hunk 2]
```diff
@@ -323,24 +323,37 @@ class Analyzer {
...
 /// the `dartfmt` from SDK 1.9 and greater.
 class DartFmt {
-  /// Run the `dartfmt` command with the `--overwrite` option. Format any files
-  /// in place.
+  /// Run the `dartfmt` command with the `--overwrite` option. Format a file, a
+  /// directory or a list of files or directories in place.
   static void format(fileOrPath) {
-    if (fileOrPath is File) fileOrPath = fileOrPath.path;
-    _run('--overwrite', fileOrPath);
+    if (fileOrPath is File) {
```
Comment (devoncarew) [inline lib/grinder_sdk.dart]: How about:

```
if (fileOrPath is File) fileOrPath = fileOrPath.path;
if (fileOrPath is! List) fileOrPath = [fileOrPath];
_run('--overwrite', fileOrPath);
```

File: tool/grind.dart [hunk 3]
```diff
@@ -68,3 +68,14 @@ Future testsBuildWeb() {
...
+
+const sourceDirectories = const [
+  'bin',
+  'example',
+  'lib',
+  'test',
+  'tool',
+  'web'
+];
+@Task('Apply dartformat oo all Dart source files')
```
Comment (devoncarew) [inline tool/grind.dart]: `oo` ==> `to` :)

**Comment by devoncarew** [comment]
Thanks! Some comments in-line.

File: lib/grinder_sdk.dart (hunk 2 above)
Comment (zoechi) [inline lib/grinder_sdk.dart]: Definitely better.

**Comment by devoncarew** [comment]
lgtm!

Will land this as is. From looking at the API for `Analyzer` though, we have `analyze` (singular)` and `analyzeFiles` (plural). https://github.com/google/grinder.dart/blob/master/lib/grinder_sdk.dart#L310

We should change DartFmt to match or relax the param type on `Analyzer.analyze` and deprecate `Analyzer.analyzeFiles`.

**Pull request merged**